docs/lab/06/index.md

```diff
@@ -0,0 +1,359 @@
+---
+sidebar_position: 7
+description: Serial Peripheral Interface
+slug: /lab/06
+---
+
+# 06 - Serial Peripheral Interface
+
+This lab will teach you how to communicate with hardware devices using the Serial Peripheral Interface (SPI) protocol, in Embassy. 
+
+## Resources
+
+TBD
+
+## Serial Peripheral Interface (SPI)
+
+The Serial Peripheral Interface (or SPI) is a communication protocol used between microcontrollers and peripheral ICs (integrated circuits). It allows synchronous transfer of data between one *main* device and one or more *sub* devices (e.g. sensors). The main device is the one that dictates the communication.
+
+### Data transmission
+
+![SPI_protocol](images/spi_network.svg)
+
+During SPI transmission, there are 4 wires used:
+
+- **CLK** - clock - generated by the main device - subs use it to synchronously read and write to the bus
+- **MOSI** - **M**ain **O**ut **S**ub **I**n - used to carry data from the main device to *all* the subs
+- **MISO** - **M**ain **I**n **S**ub **O**ut - used to carry data from the *active* sub to the main device
+- **CS** - chip select - selects the sub to be activated, or the sub that the main wants to communicate with
+                       - one wire per sub
+                       - inactive subs will disconnect from the *MOSI* and *MISO* lines
+
+:::info
+The communication is *full-duplex*. This means that, during data transmission, both the main and the sub **must write** to the MOSI/MISO wires, no matter if that data is relevant or not.
+:::
+
+#### Transmission example
+
+1. main sets `CS` to `LOW` - the sub that the main wants to communicate with is activated - all other subs are already disconnect from `MOSI` and `MISO` lines as their CS line is not active
+2. main writes the first bit on the `MOSI` line, and, *simultaneously*, sub writes the first bit on the `MISO` line
+3. main starts the clock
+4. on *rising edge* 
+    - main reads data from the `MISO` line 
+    - sub reads data from the `MOSI` line
+5. on *falling edge*
+    - main writes data to the `MOSI` line
+    - sub writes data to the `MISO` line
+6. repeat steps 4 and 5 until main stops the clock
+
+:::note
+Whether steps 4 and 5 happen on rising/falling edge depends on the `CPOL` configuration of the SPI.
+:::
+
+![SPI_example](images/spi_network_transmission.svg)
+
+In the image above, the main first exchages data with the first sub by setting its corresponding `CS` line to low. After it finishes transmission with sub 1, it deactivates it and, afterwards, initiates communication with the second sub by setting its `CS` line to low. Notice that both main and sub read and write to the MOSI/MISO line simultaneously.
+
+### SPI modes
+
+SPI has 4 different modes which define when data is read or written. These modes can be configured through 2 registers: `CPOL` and `CPHA`.
```

Comment:
For this section please take a look at https://www.analog.com/en/resources/analog-dialogue/articles/introduction-to-spi-interface.html at Clock Polarity and Clock Phase

Reply:
Fixed it.